parlai/tasks/quac/build.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+# Download and build the data if it does not exist.
+
+import parlai.core.build_data as build_data
+import os
+import json
+
+TRAIN_FILENAME = 'train_v0.2.json'
+VALID_FILENAME = 'val_v0.2.json'
+
+URL = 'https://s3.amazonaws.com/my89public/quac/'
+SHOULD = '__SHOULD__'
+MAYBE = '__MAYBE__'
+SHOULD_NOT = '__SHOULDNOT__'
+
+YES = '__YES__'
+NO = '__NO__'
+NEITHER = '__NEITHER__'
+
+MAP_CONTINUATION = {'m': MAYBE, 'f': SHOULD, 'n': SHOULD_NOT}
+MAP_AFFIRMATION = {'y': YES, 'n': NO, 'x': NEITHER}
+
+OUTPUT_FORMAT = ("text:{question}\tfollowup:{continuation}\tyesno:"
+                 "{affirmation}\tanswer_starts:{start}\tlabels:{labels}")
+
+
+def make_parlai_format(outpath, dtype, data):
+    print('building parlai:' + dtype)
+    with open(os.path.join(outpath, dtype + '.txt'), 'w') as fout:
+        for line in data:
+            for each in line['paragraphs']:
+                output = []
```

Comment:
this is a lot of indentation. It might make sense to refactor this out:

```python
def _handle_paragraph(paragraph):
    output = []
    story = # ...
    # ...
    return "".join(output)
```
and then here you'd write:

```python
for line in data:
    for paragraph in line['paragraphs']:
        fout.write(_handle_paragraph(paragraph))
```

Reply:
Thanks! that is really useful tips!